packages/core/src/components/EditableTypography/EditableTypography.tsx

```diff
@@ -138,15 +148,47 @@ const EditableTypography: VibeComponent<EditableTypographyProps, HTMLElement> =
       }
     }
 
-    function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    function handleChange(event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) {
       setInputValue(event.target.value);
+
+      if (multiline) {
+        resizeTextarea();
+      }
     }
 
     const toggleKeyboardEditMode = useKeyboardButtonPressedFunc(toggleEditMode);
 
     function focus() {
       if (inputRef.current) {
         inputRef.current?.focus();
+
+        if (multiline) {
+          calculateTextareaHeightAttrs();
+        }
+      }
+    }
+
+    /* Dynamically resizes the textarea to fit its content */
+    function resizeTextarea() {
+      if (inputRef.current) {
+        // Temporarily set the height to "auto" to accurately measure the scroll height of the content inside the textarea.
+        setInputHeight("auto");
+
+        requestAnimationFrame(() => {
```

Comment:
check layoutEffect instead?

Reply:
layout effect does the same thing, and even adds unnecessary calculations on the lineHeight
dropped it
@talkor